### ISSUE TITLE
Issues with webnative.apps.* type signatures and implementations

### ISSUE THREAD

**Issue by matheus23** [fission-suite/webnative#182]
This is what `await webnative.apps.index()` returns for me:
```json
{
    "427": [
        "elegant-small-old-wolf.fission.app"
    ],
    "1291": [
        "victorious-short-azure-witch.fission.app"
    ],
    "1292": [
        "thin-azure-ghost.fission.app"
    ],
    "1293": [
        "petite-oval-teal-tulip.fission.app"
    ],
    "1294": [
        "web-uploaded-app.fission.app"
    ],
    "1295": []
}
```

The typescript type is:
```ts
export declare type App = {
    domain: string;
};
/**
 * Get A list of all of your apps and their associated domain names
 */
export declare function index(): Promise<Array<App>>;
```

The type doesn't match what the server actually responds.

**Comment by matheus23** [comment]
Oh, and I noticed that the signature is incorrect for `webnative.apps.create()`, too:
```js
await webnative.apps.create("new-app")
// outputs "new-app.fission.app"
```
The signature of that function is `Promise<App>`, but it should be `Promise<string>`.

**Comment by matheus23** [comment]
Oh and one more: The parameter `subdomain: Maybe<string>` is also incorrect (`Maybe<string>` is a synonym for `string | null`), because calling `webnative.apps.create(null)` will run a query with the app name `"null"`.
The call does the expected thing when calling it like this: `webnative.apps.create()`. It checks for defined-ness (undefined or not), instead of for whether it's null or not.

**Comment by matheus23** [comment]
Oh yeah, and while we're at it: The current definition of `webnative.apps.create` doesn't create a valid REST url call when you specify a subdomain. It's missing a `subdomain=` in the query string.

**Comment by icidasset** [comment]
Fixed in #176
